java/java.lsp.server/nbcode/integration/test/unit/src/org/netbeans/modules/nbcode/integration/VerifySimpleTemplatesTest.java

```diff
@@ -0,0 +1,128 @@
+ * under the License.
+ */
+
+package org.netbeans.modules.nbcode.integration;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import junit.framework.Test;
+import static junit.framework.TestCase.assertTrue;
+import static junit.framework.TestCase.fail;
+import org.netbeans.api.actions.Editable;
+import org.netbeans.api.actions.Openable;
+import org.netbeans.junit.NbModuleSuite;
+import org.netbeans.junit.NbTestCase;
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.FileUtil;
+import org.openide.loaders.DataFolder;
+import org.openide.loaders.DataObject;
+
+public class VerifySimpleTemplatesTest extends NbTestCase {
+    private static Logger LOG = Logger.getLogger(VerifySimpleTemplatesTest.class.getName());
+
+    public VerifySimpleTemplatesTest(String name) {
+        super(name);
+    }
+
+    public static Test suite() {
+        String v = System.getProperty("java.version");
+        if (v == null || !v.startsWith("1.8")) {
+            return NbModuleSuite.emptyConfiguration().suite();
+        }
+
+        return NbModuleSuite.emptyConfiguration().
+            clusters("(extide|java).*").
+            enableModules(".*", ".*").
+            honorAutoloadEager(true).
+            failOnException(Level.INFO).
+            addTest(VerifySimpleTemplatesTest.class).
+            suite();
+    }
+
+    public void testAllTemplates() throws IOException {
+        clearWorkDir();
+
+        DataFolder scratch = DataFolder.findFolder(FileUtil.toFileObject(getWorkDir()));
+
+        DataFolder templates = DataFolder.findFolder(FileUtil.getConfigFile("Templates"));
+        List<DataObject> groups = new ArrayList<>();
+        quickPickTemplates(templates, groups);
+
+        List<DataObject> errorneus = new ArrayList<>();
+        int cnt = 0;
+        for (DataObject g : groups) {
+            assertTrue("It is a folder: " + g, g instanceof DataFolder);
+            DataFolder f = (DataFolder) g;
+
+            List<DataObject> simpleTemplates = new ArrayList<>();
+            quickPickTemplates(f, simpleTemplates);
+
+            for (DataObject t : simpleTemplates) {
+                LOG.log(Level.WARNING, "Processing {0}", t.getPrimaryFile().getPath());
+                final DataObject generated = t.createFromTemplate(scratch, "Test" + ++cnt);
+                final FileObject pf = generated.getPrimaryFile();
+                final Editable edit = generated.getLookup().lookup(Editable.class);
+                if (edit != null) {
+                    LOG.log(Level.WARNING, "Editing {0}", pf.getPath());
+                    edit.edit();
+                } else {
+                    Openable open = generated.getLookup().lookup(Openable.class);
+                    if (open != null) {
+                        LOG.log(Level.WARNING, "Opening {0}", pf.getPath());
```

Comment:
Is this `WARNING` level message necessary in this 'successful' path - or is it meant to trace the progress in the test job logs ?

Reply:
This kind of integration IDE tests is better to be verbose, I think.